### ISSUE TITLE
Paper: Parameters' metrics in the Bayesian framework

### ISSUE THREAD

**Issue by DominiqueMakowski** [easystats/bayestestR#11]
As you might have seen, one of the [vignette](https://easystats.github.io/bayestestR/articles/3_IndicesExistenceComparison.html) is a bit more detailed and looks almost like a paper. Actually, the whole bayestestR package came from a paper that I started writing to compare different indices together, but the implementation (in psycho) was a bit messy and after some preliminary reviews, I've decided to reimplement the functions in a separate and dedicated package, and ended up mostly copying the manuscript to the vignettes.

However, I would still like to try to publish some sort of guide and comparison of indices (including the comparison with the frequentist *p* value, about which almost every *already-convinced-Bayesian* says "it's doesn't make sense" but probably is one of the things that many frequentists interested in switching to the Bayesian framework would like to see to feel "comfy" and secure - giving them some references and benchmarks).

Anyway, the current version tries to see how the different "indices of existence" are influenced by different things such as sample size, noise, priors specification and so on. The end goal was to give a "intuitive" feeling of these metrics to users and neophytes and draw reporting guidelines (that would be then implemented in `report` 😏 ). However, one of the thing that could be improved is the generalizability, as the current results are based on a simulation of a simple regression model. But I am not sure what is the most smart way of extending the analysis to different models to see if and how it influence sthe metrics.

Would you be interested in such collaborating around such paper? Since I'd like to try a continuous and open writing process, through the progressive enhancement of the vignettes, there is absolutely no time pressure of any kind. Once we would be happy with the vignettes, we could try submitting its content to a journal. What are your thoughts?

**Comment by strengejacke** [comment]
That sounds good! I'm always open to and happy about writing a paper on Bayesian analysis, because I would like to have some "citeable" references. :-)

I think, though, that such a paper would benefit from a wording that does not overhype the Bayesian philosophy. There's a very interesting discussion here:
https://discourse.datamethods.org/t/language-for-communicating-frequentist-results-about-treatment-effects/934/16

and an important comment later on here:
https://discourse.datamethods.org/t/language-for-communicating-frequentist-results-about-treatment-effects/934/28

> Frequentist results are chronically misrepresented and hence miscommunicated. Yes their Bayesian counterparts can be phrased more straightforwardly, but that’s not always an advantage because their overconfident descriptions are harder to see as misleading. “_The_ posterior probability” is a prime example because there is no single posterior probability, there is only _a_ posterior probability computed from a possibly very wrong hierarchical (prior+data) model. Similarly there is no single P-value for a model or hypothesis.

**Comment by strengejacke** [comment]
Though this is "just" another opinion from Sander, I find it convincing and see the point that just using Bayesian methods does not overcome all existing problems.

**Comment by DominiqueMakowski** [comment]
this is very interesting, I am indeed guilty of overhyping it 😅

**Comment by strengejacke** [comment]
It was more a general comment on overhyping, not specifically addressing you ;-)
But I think this is in general not uncommon that when you switch from frq to Bayes, you see all the benefits, and you start seeing _only_ benefits (at least this was what I observed for me personally).

**Comment by DominiqueMakowski** [comment]
definitely... Sander's comments are quite refreshing, it's rare to read well-informed and provoking takes on frequentist indices

**Comment by DominiqueMakowski** [comment]
I think that decomposing the paper/vignette in 3 could be clearer than modulating and assessing the effect of everything at once.

I was thinking about 3 "experiments" (or parts):

- one in which are modulated effect size, sample size and noise, in which we compare the indices and their relationship with frequentist ones. For this part, the generation script is pretty much in shape. I need to update the data for the vignettes by re-running it again(it's in [circus](https://github.com/easystats/circus/blob/master/vignettes/bayesian_indices.Rmd), I'd be glad to have your input), which takes about a week on a server 😞. I will start it right after I finish with my current scripts for a paper revision (which is due next week).
- a second experiment to investigate the sensitivity of indices to priors specification. I am not sure yet of the best way to investigate that. I was thinking of generating data with 3 different types of priors, located at 0, at the "true" effect (although defining this is a bit of a pickle) and at the inverse of the true effect. And see how it affects the indices.
- a third experiment could test the robustness to sampling quality. One way to test that could be modulating the sampling characteristics such as the number of chains, burn-in, draws, algorithm etc.

**Comment by DominiqueMakowski** [comment]
I finally am done with a personal analysis. So I am planning to re-run the [script]() to obtain the data for the [indices of effect existence](https://easystats.github.io/bayestestR/articles/indicesExistenceComparison.html) vignette (which will take about 1 week).

I am planning to add 2 indices for each coef's posterior:

- BF - savage dickey with prior `rcauchy_perfect` with location 0 and SD = SD of the posterior 
- BF - savage dickey with prior `rcauchy_perfect` with location 0 and SD = 1

Mainly out of curiosity, to see how it relates to other indices such as ROPE and so on.

[here](https://github.com/easystats/circus/blob/1f70a7b066f3338964cc0bf1d46dee3f0312586f/vignettes/bayesian_indices.Rmd#L111) is the code used to extract the relevant indices. Let me know if you have any suggestions about other indices.

**Comment by mattansb** [comment]
@DominiqueMakowski I am also interested to see how these heuristic priors work out in the new BF-SD. Please keep me posted!

**Comment by mattansb** [comment]
@DominiqueMakowski As `stanreg` models are now supported by the Savage-Dickey BF function, which extracts actual priors directly, maybe you could add that to the vignette as well!

(I feel very at home here, as you can tell by how comfortably I allow myself to suggest some extra work for you 😄 )

**Comment by DominiqueMakowski** [comment]
Yes, I think I'll replace the current two BFs (Cauchy 0.707 and Cauchy 1) by Cauchy 0.707 and model's priors.

**Comment by mattansb** [comment]
I was looking at the Comparison of Indices of Effect Existence, and had some thoughts regarding the BFs:
1. Am I correct to assume that the prior scales are standardized? 
2. I think it would be interesting to have 2*2 bayes factors:
    1. Model comparison BF (effect model vs intercept only model) and Savage-Dickey (posterior vs prior).
    2. Prior `~cauchy(0, 0.707)` / `~student_t(3, 0)` (autoscaled) - these seem to be the two most popular options.
3. I find it very weird that when the null is true, the log(BF) never goes bellow -2.5, even as sample size increases and noise is low, which is inconsistent with other works. On that note...
4. Unless I am reading the vignette wrong (which is totally possible), none (or at least most?) of the indices seem to be informative when the null is true (all seem to behave similar to the p-value) which is disapointing, as the main draw to Bayesian stats is often the ability to support the null. Does only the ROPE+equi allow for this?

**Comment by DominiqueMakowski** [comment]
For the BF part:
- this vignette takes focus in the characteristics on the parameters only, not the entire model (thus model comparison model would be for another vignette :)
- due to our recent changes and direct support of sdBF for rstanarm, I'd say it's best if we keep only this BF the way it is computed by bayestestR

For other weirdnesses in the data, it needs to be re-generated anyway, so I would mainly focus on making sure the [generating script](https://github.com/easystats/circus/blob/master/vignettes/bayesian_indices.Rmd) is ok (as it takes a long time to run, so we have to make sure it works as expected :).

**Comment by DominiqueMakowski** [comment]
@strengejacke @mattansb 
I have reformatted the code related to the simulation study: see [here](https://easystats.github.io/circus/articles/bayesian_indices.html). 

Please try to have a look if you find the time (I know it's not a page-turner), I would like to have your opinion before running the simulation on the server.

While I think that study 1 (comparison with frequentist models and effect of sample size and error) and study 2 (effect of sampling characteristics) are "working" (although let me know if the boundaries for the parameters tested make sense for you), I have a problem with study 3; which aims at investigating the effect of prior specification.

In my opinion, I think it would be interesting to compare 3 types of priors, the default (uninformative, centred around 0) with a congruent prior (in the sense of the effect) and an incongruent (in the opposite of the effect's direction). For the linear models, this is not a problem. The true effect is either 0 or 1 (roughly corresponding to a perfect correlation). Thus, in the case a true effect is 1, a congruent prior is normal(1, 1) and an incongruent prior is normal(-1, 1). 
But things get more complicated in the case of logistic models, as the actual coefficient of a "perfect" correlation is quasi-undefined (but it is clearly higher than 1):
![image](https://user-images.githubusercontent.com/8875533/58031045-3815d380-7b52-11e9-91b0-68f1960cbbf1.png)

In this context, how to specify an "informative" prior so that it is comparable (in terms of magnitude) between linear and logistic models? Any ideas?

**Comment by mattansb** [comment]
Some notes / questions:

1. I think that the `params$BF_cauchy07` part is kind of cheating - you're not *really* specifying the prior on which the posterior is based here, you're just "feeding" sdBF a distribution to compare against, which might give bogus results:
```R
params$BF_cauchy07 <- as.numeric(bayesfactor_savagedickey(posterior,
                                    prior = distribution_cauchy(length(posterior),
                                                                location=0,
                                                                scale=0.707),
                                    hypothesis = 0,
                                    verbose=FALSE))
```
2. The code for the actual sdBF can be simplified (shouldn't add much processing time considering such small Ns / simple models) to:
```R
params$BF <- bayesfactor_savagedickey(models$bayesian)$BF[2]
```
3. Unless I'm reading the code wrong - is the max noise set at 0.66? If so, this corresponds to *r = 0.8*, which is quite large. I would go to maybe max error/noise of 5 (for *r = 0.15*) maybe even more?

Other than that, code seems alright :)

**Comment by DominiqueMakowski** [comment]
@mattansb 

1. You're right, I'll remove this BF to only use the one using the model's priors.
2. I would have thought so too :) Unfortunately, it doesn't work if I pass the model directly... Nore sure why, but my guess is that it can't retrieve the model properly as the model is passed too deeply and is not accessible from the global scope 😕 
3. The noise is actually from 0.33 to 6.66, I am not sure where you got the 0.66 value 🤔

**Comment by mattansb** [comment]
> 3. The noise is actually from 0.33 to 6.66, I am not sure where you got the 0.66 value 🤔

Oops, I seem to have seen that first 6 as a 0...

**Comment by strengejacke** [comment]
When would you like to have feedback on this?

**Comment by DominiqueMakowski** [comment]
whenever you can, no pressure, it is not urgent 
(though needed to generate the data for the vignettes/paper) ;)

**Comment by mattansb** [comment]
A minor change, in plotting: I it would be good to add marginal density plots / histograms to the plots [comparing all the measures to frequentist p-values](https://easystats.github.io/bayestestR/articles/indicesExistenceComparison.html#relationship-with-the-frequentist-p-value). This will show that the Bayesian measures are consistent when H0 is true (i.e., they converge), which is not the case for the p-values (that converge when H1 is true, but have a uniform distribution when H0 is true).

**Comment by DominiqueMakowski** [comment]
Good point! If you know how to nicely add the marginal densities, could you add them to the vignette?

**Comment by mattansb** [comment]
I do not... I this this can be done with [`ggExtra`](https://cran.r-project.org/web/packages/ggExtra/README.html), though I've never used it before... and I don't know how it deals with faceting...

**Comment by DominiqueMakowski** [comment]
Okay, I'll have a look once we have the new data. Let's add this in a list so we don't forget:

- [ ] add marginal densities to p-value comparison

**Comment by DominiqueMakowski** [comment]
https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007128
https://manubot.org/

👀

**Comment by DominiqueMakowski** [comment]
I updated the new data, and recompiled the vignette... but the size of the package went up (above CRAN's limit) due to the images and all no matter how I tried, so in the end I moved the draft [here](https://github.com/DominiqueMakowski/publications/tree/master/makowski_2019_bayesian/manuscript) for now. 

Also, after re-reading it all, I think it would be better to stick with "study 1" (where we modulate the sample size and the noise), there will already quite a lot of things to discuss. 

Once this paper it out, we might summarise it in the vignette and redirect to the full paper ☺️

**Comment by strengejacke** [comment]
How do we compile the RMD to have a "readable" version for the paper, and how should we edit?

**Comment by DominiqueMakowski** [comment]
I have knitted a readable version in word (using frontiers template) but indeed we can reknit it in a md or pdf format it will be easier to check

**Comment by mattansb** [comment]
Alright - how to you wish to proceed? 

(Or: how can I comment? 😅)

For now I'm writing my comments here, so I don't forget them:

1. Figure 4 (Relationship with frequentist p-based arbitrary clusters) - I would make two changes:
    1. Split the figure in to two: one where there is an effect and another where there isn't - this would give some mapping for the indices onto power / false discovery.
    2. Combine all the alpha thresholds into one plot with multiple s-curves:
![image](https://user-images.githubusercontent.com/35330040/61872622-7e8a0480-aeec-11e9-8b2f-b88b32a3b8fb.png)

2. I think the "noise" parameter should be followed with the equivalent "effect size", which I think people are more use to thinking about.

**Comment by mattansb** [comment]
Are we going with clone and push?

**Comment by DominiqueMakowski** [comment]
Can you guys fork the other repo and pull there?

**Comment by DominiqueMakowski** [comment]
If not, it might be easier to createand make more sense to create a devoted repo for that, for instance under easystats/easystats/publications or something along those lines, where we would store papers and data related to the project?

**Comment by mattansb** [comment]
> under easystats/easystats/publications

I think would be easier for collab

**Comment by DominiqueMakowski** [comment]
I agree, I'll transfer

**Comment by DominiqueMakowski** [comment]
[moved](https://github.com/easystats/easystats/tree/master/publications/makowski_2019_bayesian)
